### ISSUE TITLE
3PID invites don't work any more.

### ISSUE THREAD

**Issue by ara4n** [vector-im/riot-web#13430]
To reproduce:
 * Invite an email for a user who isn't on Matrix to a room
 * User gets an email inviting them to sign up on Riot
 * User signs up on Riot; so far so good.
 * User has to verify their email, despite answering a 3PID invite in the first place, which feels a bit redundant but not the end of the world.
 * 3PID invite isn't converted to a normal invite.
 * If you try to re-invite the user again, it sends them another 3PID invite.  It's like their signup was never bound.

**Comment by turt2live** [comment]
If you share the email after reusing the same tab, it'll figure it out. Otherwise it largely seems to expect multiple tabs to be used.

**Comment by dbkr** [comment]
So I think there's a whole collection of problems here, but possibly the root of the problem is that once you're done signing up, we now don't log you in straight away, instead telling you to sign in, and we don't persist the invite intent after you signed in, so we dump you in an empty client rather than prompting you to join the room you were invited to. In which case the fix would be to show the room you were invited to after you sign in, although it's still a complex process.

As Travis says, the 3pid invite won't come through as a regular invite unless you go into settings and share the email (clicking a third email link!) because we don't publish mappings to ISes by default in a post-privacy world.

I'm not sure what bearing re-using tabs has on anything.

**Comment by ara4n** [comment]
This is biting some VIPs.

**Comment by jryans** [comment]
@ara4n @neilisfragile At the moment, we're a bit stuck on this one, because as @dbkr suggests above, it's unclear what change we actually want to make here. I'll leave this in the fire column for now until Neil returns so we remember to discuss it.

**Comment by neilisfragile** [comment]
More context on how to reproduce the bug via @ara4n and @nadonomy 

"If you reuse the tab for both invite and validation in signup then it all breaks and if you then “go back to the app” by copy pasting the link from the invite email, you can’t login any more"

**Comment by turt2live** [comment]
I can't reproduce this, and am unclear what the actual root issue is here. We seem to have made Element remember the room ID you were looking at, though it can disappear with ease. I've flagged https://github.com/vector-im/element-web/issues/15131 to cover this case.

Some reports are that a login link isn't working, but I've been unable to reproduce any significant problems on 3 different stacks (local, matrix.org, custom). 

The overall flow is pretty terrible, which might be the bug here. I've tracked this as https://github.com/vector-im/element-web/issues/15130 pending design to triage.

**Comment by turt2live** [comment]
Right, so this issue is complicated but boils down to us losing state due to people pasting the identity server verification link over top of the app. Fixing that properly leads into some cobwebs around the auth system, so as a quick win we're going to fix this for the affected VIPs using alternative methods (if possible) and look at resolving this properly alongside the numerous other 3pid invite issues.

**Comment by turt2live** [comment]
For others trying to fix this: the solution is to run 3PID invites through the homeserver rather than the identity server. For Synapse this means not using `account_threepid_delegates` and instead setting up the email configuration (including `client_baseurl`) so it can send things. Sydent can still be used to store the invite, though https://github.com/vector-im/element-web/issues/15131 and https://github.com/vector-im/element-web/issues/15134 apply.
